docs/05_sdk_api.adoc

```diff
@@ -0,0 +1,812 @@
+)
+val annotations = listOf("namespace-my-annotation")
+
+client.fhir4.fetch(
+    "recordId",
+    updatedFhir4Observation,
+    annotations,
+    object : Callback<Fhir4Record<Fhir4Resource>> {
+        override fun onError(exception: D4LException) {
+            // Exception
+        }
+        override fun onSuccess(result: Fhir4Record<Fhir4Resource>) {
+            // Created record
+        }
+    }
+)
+----
+
+===== Data
+
+[source,kotlin]
+----
+fun fetch(
+    recordId: String,
+    callback: Callback<DataRecord<DataResource>>
+): Task
+----
+
+Example:
+
+[source,kotlin]
+----
+val updatedDataResource = DataResource("data".toByteArray())
+val annotations = listOf("namespace-my-annotation")
+
+client.data.update(
+    "recordId",
+    updatedDataResource,
+    annotations,
+    object : Callback<DataRecord<DataResource>> {
+        override fun onError(exception: D4LException) {
+            // Exception
+        }
+        override fun onSuccess(result: DataRecord<DataResource>) {
+            // Created record
+        }
+    }
+)
+----
+
+==== Delete
+
+Delete a record with a given record ID.
+
+===== FHIR 3
+
+[source,java]
+----
+public void deleteRecord(String recordId, Callback listener)
+----
+
+Example:
+
+[source,java]
+----
+sdk.deleteRecord("recordId", new Callback() {
+    @Override
+    public void onSuccess() {
+        // Record deleted
+    }
+
+    @Override
+    public void onError(D4LException exception) {
+        // Exception
+    }
+});
+----
+
+===== FHIR 4
+
+[source,kotlin]
+----
+fun delete(
+    recordId: String,
+    callback: Callback<Boolean>
+): Task
+----
+
+Example:
+
+[source,kotlin]
+----
+client.fhir4.delete(
+    "recordId",
+    object : Callback<Boolean> {
+        override fun onError(exception: D4LException) {
+            // Exception
+        }
+        override fun onSuccess(result: Boolean) {
+            // Created record
+        }
+    }
+)
+----
+
+===== Data
+
+[source,kotlin]
+----
+fun delete(
+    recordId: String,
+    callback: Callback<Boolean>
+): Task
+----
+
+Example:
+
+[source,kotlin]
+----
+client.data.delete(
+    "recordId",
+    object : Callback<Boolean> {
+        override fun onError(exception: D4LException) {
+            // Exception
+        }
+        override fun onSuccess(result: Boolean) {
+            // Created record
+        }
+    }
+)
+----
+
+=== Queries
+
+Query options are `search` and `count` with optional `annotations` to fine tune the result.
+
+==== Search
+
+Search for records with following filter options:
+
+* resourceType - class type of the searched resource (not available for data)
+* annotations - custom annotations that had been added to the records (optional)
+* startDate - the filtered records have a creation date after the start date
+* endDate - the filtered records have a creation date before the endDate
+* pageSize - define the result page size, how many records should be returned
+* offset - the offset of the result list
```

Comment:
This is not true anymore - see #179

Reply:
the documentation update is only scoped to the currenty released version

Reply:
Ah you are right...GitHub still does not respect the merge order.